src/main/java/sorald/processor/InterruptedExceptionProcessor.java

```diff
@@ -32,13 +31,42 @@ protected void repairInternal(CtCatch element) {
         CtInvocation<?> secondInvocation =
                 factory.createInvocation(firstInvocation, interruptMethod.getReference());
 
-        CtStatement statementToInsert =
-                mustTypeCheckCatchVariable(element)
-                        ? wrapInTypeCheck(element.getParameter(), secondInvocation)
-                        : secondInvocation;
+        if (mustTypeCheckCatchVariable(element)) {
+            wrapInCatcher(element, secondInvocation);
+        } else {
+            element.getBody()
+                    .addStatement(lastSafeInterruptIndex(element.getBody()), secondInvocation);
+        }
+    }
+
+    private static void wrapInCatcher(CtCatch violatedCatch, CtStatement statementToWrap) {
+        Factory factory = violatedCatch.getFactory();
+        CtTypeReference<?> refToInterruptedException =
+                factory.Type().get(InterruptedException.class).getReference();
+
+        // Remove InterruptedException from the catch
+        CtCatchVariable<?> catchVariable = violatedCatch.getParameter();
+        catchVariable.removeMultiType(refToInterruptedException);
+
+        // Add statement into a cloned catch block
+        // Currently, we clone the body of the violated catch which is theoretically wrong because
+        // there could be statements concerning only one of the exception types.
+        // Example - `if (e instanceof IOException) { ... }`. However, it is safe // NOSONAR:S125
+        // to ignore this case because it is rare.
```

Comment:
I have added the comment here we noted yesterday.

Sonar would not let me comment `if (e instanceof IOException) { ... }` inside a meaningful comment and that's why I added a comment on line 54.